### PR TITLE
replace u_scanner with DECORATE compatible version

### PR DESCRIPTION
**Pull request by rfomin**
Preparation for MBF25 DECORATE implementation, which requires more sophisticated scanner. Adapted from https://bitbucket.org/Blzut3/declarate/src/master/src/scanner.cpp

* Small refactoring of UMAPINFO.

The KEX port's `masterlevels.wad` has a different map order than our `base/masterlevels.wad/umapdef.lmp`. Should we remove all UMAPDEF at this point?

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -64,6 +64,7 @@ set(WOOF_SOURCES
     m_input.c              m_input.h
     m_io.c                 m_io.h
     m_json.c               m_json.h
+    m_scanner.c            m_scanner.h
```
Comment (fabiangreffrath) [inline src/CMakeLists.txt:67]: Please update README.md as well.

**Comment by fabiangreffrath** [comment]
> The KEX port's `masterlevels.wad` has a different map order than our `base/masterlevels.wad/umapdef.lmp`. Should we remove all UMAPDEF at this point?

But this is due to another update to the WAD files, and not related to the parser changes proposed here, right?

**Comment by rfomin** [comment]
> But this is due to another update to the WAD files, and not related to the parser changes proposed here, right?

Yes, it's not related to this PR, I'm just starting to refactor UMAPINFO a bit more locally. I think we should remove the whole UMAPDEF system at this point: popular versions of nerve.wad and masterlevels.wad now have proper UMAPINFO, also we have the woof.pk3 system now, so load order is not such a problem.
The UMAPINFO refactoring would be a separate PR.

File: src/CMakeLists.txt (hunk 1 above)
Comment (rfomin) [reply src/CMakeLists.txt:67]: Done in fb64425

**Comment by fabiangreffrath** [comment]
Alright, please proceed.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**